### PR TITLE
Adding compilation database to JSON in tu_collector

### PR DESCRIPTION
**Pull request by bruntib**

### CODE REVIEW

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tools/tu_collector/tu_collector/tu_collector.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/codechecker/codechecker/pullRequest?prid=4560555&bid=15336110)

File: tools/tu_collector/tu_collector/tu_collector.py [hunk 1]
```diff
@@ -281,6 +281,10 @@ def zip_tu_files(zip_file, compilation_database, write_mode='w'):
 
     add_sources_to_zip(zip_file, tu_files)
 
+    with zipfile.ZipFile(zip_file, 'a') as archive:
```
Comment (csordasmarton) [inline tools/tu_collector/tu_collector/tu_collector.py]: Please extend our test suite to check that the zip file contains the `compilation_database.json` file:
https://github.com/Ericsson/codechecker/blob/master/tools/tu_collector/tests/tu_collector_test.py#L53

Comment (csordasmarton) [inline tools/tu_collector/tu_collector/tu_collector.py]:
```suggestion
    with zipfile.ZipFile(zip_file, write_mode) as archive:
```

Comment (csordasmarton) [inline tools/tu_collector/tu_collector/tu_collector.py]: Also we open the zipfile multiple times if some errors occurred:
https://github.com/Ericsson/codechecker/blob/e09c190e54cb6e70eb1e86f1d538adbc27cb69f8/tools/tu_collector/tu_collector/tu_collector.py#L278-L280

We can open it only once:
```py
with zipfile.ZipFile(zip_file, write_mode) as archive:
    archive.writestr('compilation_database.json',
                     json.dumps(compilation_database, indent=2))

    if error_messages:
        archive.writestr(no_sources, error_messages)
```

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tools/tu_collector/tu_collector/tu_collector.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/codechecker/codechecker/pullRequest?prid=4560555&bid=15336110)

**Pull request merged**